### DISCUSSION TITLE
Working perfectly with DELL U4021QW + M1 Max MacBook Pro

### DISCUSSION THREAD

**Discussion by FedorSosnin** [MonitorControl/MonitorControl#945] [General]
Just wanted to say thank you for this amazing app. It's quite unbelievable that you can spend over a thousand dollars on a display and not have the volume be controllable via your Audio keys. I dont know the technical aspect of this but the user experience is just ruined with this tiny detail. 

This app is simple, clean, and professional. Well done team! You should be super proud.

**Comment by waydabber** [comment]
Thank you! :)

**Comment by johnchurchill** [comment]
> Just wanted to say thank you for this amazing app. It's quite unbelievable that you can spend over a thousand dollars on a display and not have the volume be controllable via your Audio keys. I dont know the technical aspect of this but the user experience is just ruined with this tiny detail.
> 
> This app is simple, clean, and professional. Well done team! You should be super proud.

@FedorSosnin , are you using your Dell monitor with the HDMI port on the Macbook Pro, or through an adapter? I'm finding that volume control doesn't work and brightness is iffy on the HDMI port, but both work fine if I use an Anker USB-C/HDMI adapter. I have a Dell S3221QS.

**Reply by FedorSosnin** [reply]
No. I'm using a single USB-C cable to my M1 Max MBP. It's been perfect with that setup. I haven't tried the HDMI mostly because a single cable, at least for me personally, is a huge win. I have keyboard, backup drive, and a webcam all hooked into the monitor and all of it just works. And now, I can actually use the volume and brightness controls with the built in monitor speakers.

Only downside now, is that there is no current driver for the RJ45 Ethernet adapter. So, I get half speeds when the ethernet is plugged directly into the display. I've tested it with a Ethernet to USB dongle and get full speed even if the dongle is hooked into the display via USB. So, it's literally just the ethernet plug's drivers at fault here.

**Reply by the0neyouseek** [reply]
Hi @johnchurchill 
The HDMI port on the new M1 Macbook (and  Mac mini) do not allow for DDC hardware control. So the app uses alternate methods like gamma table manipulation to control brightness. The app only support brightness control in software mode hence why you can't control volume.

**Reply by johnchurchill** [reply]
Thanks for the response, that explains it. I assume the HDMI port on my Dell 5480 (corporate controlled laptop) does not have that restriction (or Dell has a way around it) since the Dell Display Manager control sliders are changing the monitor's brightness natively.
The software brightness control you describe on the Mac is really not usable, no matter what I set the Dell monitor brightness to. At some points the monitor goes nuts with static and I have to reboot the Mac to get video back.
But that's OK, I am happy using the Anker adapters (about $15 each), which I have had for years and gives me 4K 60hz on both the Dell and all recent Macs through the USB-C ports.
Thanks for making this great app. I'm glad I found it.
